ee/server/service/mdm.go

```diff
@@ -1055,6 +1055,62 @@ func (svc *Service) GetMDMDiskEncryptionSummary(ctx context.Context, teamID *uin
 	}, nil
 }
 
+func (svc *Service) mdmAppleEditedMacOSUpdates(ctx context.Context, teamID *uint, updates fleet.MacOSUpdates) error {
+	// TODO: is there a notion of "DDM enabled" or not, where the DDM profile
+	// should not be created?
+
+	if updates.MinimumVersion.Value == "" {
+		// OS updates disabled, remove the profile
+		if err := svc.ds.DeleteMDMAppleDeclarationByName(ctx, teamID, mdm.FleetMacOSUpdatesProfileName); err != nil {
+			return err
+		}
+		var globalOrTeamID uint
+		if teamID != nil {
+			globalOrTeamID = *teamID
+		}
+		if err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, []uint{globalOrTeamID}, nil, nil); err != nil {
+			return ctxerr.Wrap(ctx, err, "bulk set pending host profiles")
+		}
+		return nil
+	}
+
+	// OS updates enabled, create or update the profile with the current settings
+
+	const macOSSoftwareUpdateType = `com.apple.configuration.softwareupdate.enforcement.specific`
+	ident := uuid.NewString()
+	// TODO(mna): is that correct payload? Identifier is a uuid?
+	rawDecl := []byte(fmt.Sprintf(`{
+	"Identifier": %q,
+	"Type": %q,
+	"Payload": {
+		"TargetOSVersion": %q,
+		"TargetLocalDateTime ": "2024-03-01T12:00:00,"
+	}
+}`, ident, macOSSoftwareUpdateType, updates.MinimumVersion.Value))
+	d := fleet.NewMDMAppleDeclaration(rawDecl, teamID, mdm.FleetMacOSUpdatesProfileName, macOSSoftwareUpdateType, ident)
+
+	// associate the profile with the built-in label that ensures the host is on
+	// macOS 14+ to receive that profile
+	lblIDs, err := svc.ds.LabelIDsByName(ctx, []string{fleet.BuiltinMacOS14PlusLabelName})
+	if err != nil {
+		return err
+	}
+	d.Labels = []fleet.ConfigurationProfileLabel{
+		{LabelName: fleet.BuiltinMacOS14PlusLabelName, LabelID: lblIDs[fleet.BuiltinMacOS14PlusLabelName]},
+	}
+
+	decl, err := svc.ds.NewMDMAppleDeclaration(ctx, d)
```

Comment:
TODO (follow-up PR): prevent creating/deleting/returning this Fleet-reserved profile name (I think creating/deleting is already prevented because the name was added to the fleet reserved list, but there's testing to add around that and definitely missing code so it doesn't get removed on batch-apply profiles).

Reply:
And also filter-out in counts and filters of profiles by status.

Reply:
basically, must be treated the same way we do for Windows OS Updates (which is also via a hidden Fleet-controlled profile).

Reply:
> TODO (follow-up PR): prevent creating/deleting/returning this Fleet-reserved profile name (I think creating/deleting is already prevented because the name was added to the fleet reserved list, but there's testing to add around that and definitely missing code so it doesn't get removed on batch-apply profiles).

Linking to this [PR](https://github.com/fleetdm/fleet/pull/18043) which may give a jump start on the testing piece for batch-apply.